### ISSUE TITLE
Cannot modified established architecture shape.

### ISSUE THREAD

**Issue by AndreGuerra123** [keras-team/keras#10923]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hello,
I collected several models in the .json format which I import using model_from_json function.
These models have different inputs and outputs (tensors) in the respective first and final layers. (Some of these models are not Sequential). In my API the user can select the input_shape and data_format coming from the flow generator. So, I need to modify the architecture shape, in order to take data in the format the user specified. However, after spending the two days looking for alternative ways I could not find an effective way of changing the architecture input layer. Keras supposed to be intuitive. xP Hereby, I'll give an example of a small model architecture generated by me and some ideas I've tried. Honestly, I really appreciate some help with this.

Model:
```
model = Sequential()
model.add(Conv2D(32, (3, 3), padding='same', input_shape = (100,100,3)))
model.add(Activation('relu'))
model.add(Conv2D(32, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(64, (3, 3), padding='same'))
model.add(Activation('relu'))
model.add(Conv2D(64, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(2))
model.add(Activation('softmax'))
```

Goal:
Make it receive batch data in the shape (None,32,32,3), 'channels_last'
Make it output batch data in the shape (None,6)

Attempted:
1. Using set shape
`model.layers[0].input.set_shape((None,32,32,3))` 
2. Using config
```
layer = model.layers[0]
config = layer.get_config()
config['input_shape']=(32,32,3)
config['batch_input_shape']=(None,32,32,3)
model.layers[0] = layers.deserialize({'class_name': layer.__class__.__name__, 'config': config})
model.inputs[0] = keras.layers.Input(shape=(32,32,3))
```
3. Changing tensor directly
def changeInputsShape(model, inputs):
   assert len(model.inputs) == len(inputs), "Model inputs and assigned inputs mismatch."

   for i in range(len(inputs)):
     model.inputs[i] = keras.layers.Input(shape=inputs[i])

Reason:
1. set_shape only update non defined dimensions of the previously defined tensor shape.
2. setting configuration does not update input tensors themselves, compiles not run.
3. with model.build() prior to compiling: `Graph disconnected: cannot obtain value for tensor Tensor("conv2d_1_input:0", shape=(?, 100, 100, 3), dtype=float32) at layer "conv2d_1_input". The following previous layers were accessed without issue: []`
3. with compiling: `Could not compute output Tensor("activation_6/Softmax:0", shape=(?, 2), dtype=float32)`

Note: This is note a weights/ knowledge tranfer problem since the model is not trained so it should be relatively simpler. I guess not.

Any help would be appreciated,
Andre
